core/src/main/java/net/opentsdb/query/ReadCacheQueryPipelineContext.java

```diff
@@ -152,6 +152,22 @@
         break;
       }
       
+      if (config instanceof TimeSeriesDataSourceConfig) {
+        final List<String> types = ((TimeSeriesDataSourceConfig) config).getTypes();
+        if (types != null && !types.isEmpty()) {
+          for (final String type : types) {
+            if (!type.toUpperCase().equals("METRIC")) {
```

Comment:
why not equalsIgnoreCase and isn't Types an enum??

Reply:
oh yeah, I can do that. No types is a string since we don't know what we'll have in the future (traces, logs, etc.)

Reply:
Oh nvm!!